runtime/src/snapshot_utils.rs

```diff
@@ -649,64 +896,280 @@ pub fn bank_from_snapshot_archive<P>(
+#[cfg(test)]
+#[allow(clippy::too_many_arguments)]
+fn bank_from_incremental_snapshot_archive<P, Q>(
+    account_paths: &[PathBuf],
+    frozen_account_pubkeys: &[Pubkey],
+    snapshot_path: &Path,
+    full_snapshot_archive_path: P,
+    incremental_snapshot_archive_path: Q,
+    archive_format: ArchiveFormat,
+    genesis_config: &GenesisConfig,
+    debug_keys: Option<Arc<HashSet<Pubkey>>>,
+    additional_builtins: Option<&Builtins>,
+    account_secondary_indexes: AccountSecondaryIndexes,
+    accounts_db_caching_enabled: bool,
+    limit_load_slot_count_from_snapshot: Option<usize>,
+    shrink_ratio: AccountShrinkThreshold,
+    test_hash_calculation: bool,
+) -> Result<(Bank, BankFromArchiveTimings)>
+where
+    P: AsRef<Path> + std::marker::Sync,
+    Q: AsRef<Path> + std::marker::Sync,
+{
+    do_bank_from_snapshot_archives(
+        account_paths,
+        frozen_account_pubkeys,
+        snapshot_path,
+        full_snapshot_archive_path,
+        Some(incremental_snapshot_archive_path),
+        archive_format,
+        genesis_config,
+        debug_keys,
+        additional_builtins,
+        account_secondary_indexes,
+        accounts_db_caching_enabled,
+        limit_load_slot_count_from_snapshot,
+        shrink_ratio,
+        test_hash_calculation,
+    )
+}
 
-    let mut untar = Measure::start("snapshot untar");
-    let divisions = std::cmp::min(
+/// Rebuild a bank from snapshot archives.  Handle either just a full snapshot, or both a full
+/// snapshot and an incremental snapshot.
+#[allow(clippy::too_many_arguments)]
+fn do_bank_from_snapshot_archives<P, Q>(
+    account_paths: &[PathBuf],
+    frozen_account_pubkeys: &[Pubkey],
+    snapshot_path: &Path,
+    full_snapshot_archive_path: P,
+    incremental_snapshot_archive_path: Option<Q>,
+    archive_format: ArchiveFormat,
+    genesis_config: &GenesisConfig,
+    debug_keys: Option<Arc<HashSet<Pubkey>>>,
+    additional_builtins: Option<&Builtins>,
+    account_secondary_indexes: AccountSecondaryIndexes,
+    accounts_db_caching_enabled: bool,
+    limit_load_slot_count_from_snapshot: Option<usize>,
+    shrink_ratio: AccountShrinkThreshold,
+    test_hash_calculation: bool,
+) -> Result<(Bank, BankFromArchiveTimings)>
+where
+    P: AsRef<Path> + std::marker::Sync,
+    Q: AsRef<Path> + std::marker::Sync,
+{
+    let parallel_divisions = std::cmp::min(
         PARALLEL_UNTAR_READERS_DEFAULT,
         std::cmp::max(1, num_cpus::get() / 4),
     );
-    let unpacked_append_vec_map = untar_snapshot_in(
-        &snapshot_tar,
-        unpack_dir.as_ref(),
+
+    let (
+        _full_snapshot_unpack_dir,
+        full_snapshot_unpacked_snapshots_dir,
+        full_snapshot_unpacked_append_vec_map,
+        full_snapshot_version,
+        full_snapshot_measure_untar,
+    ) = do_snapshot_unarchive_preparation(
+        snapshot_path,
+        TMP_SNAPSHOT_PREFIX,
+        &full_snapshot_archive_path,
+        "snapshot untar",
```

Comment:
Yeah, I wasn't sure how much to rename. See this comment: https://github.com/solana-labs/solana/pull/18504#discussion_r668915990

My thought for this PR was everything that didn't include "incremental" meant it was a full snapshot, which is the way it works today, since all snapshots are full snapshots. Then within the context of incremental snapshot code, I would always add "full" to differentiate. Whatcha think?